### PR TITLE
DM-33297: Add correction of systematic photodiode error

### PR DESCRIPTION
**Pull request by craiglagegit**
This ticket was to add the photodiode and photodiode systematic correction code to the linearizer.

### CODE REVIEW

File: python/lsst/ip/isr/photodiodeCorrection.py [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+        """
+        tableList = []
+        self.updateMetadata()
+        catalog = Table([{'PAIR': key,
+                          'PD_CORR': self.abscissaCorrections[key]}
+                         for key in self.abscissaCorrections.keys()])
+        catalog.meta = self.getMetadata().toDict()
+        tableList.append(catalog)
+
+        if self.tableData is not None:
```
Comment (czwa) [inline python/lsst/ip/isr/photodiodeCorrection.py:197]: I don't think these `LOOKUP_VALUES` are being persisted correctly, are they?  They appear here, but I don't see an associated block in `fromTable` or in `to/fromDict`.

Can you add a test (`tests/test_photodiode.py` is probably the least confusing place) that at least does round-tripping tests?

**Review by czwa — CHANGES_REQUESTED** [review]
Mostly style issues.  One concern that the `tableData` is getting lost depending on format.

File: python/lsst/ip/isr/photodiodeCorrection.py [hunk 2]
```diff
@@ -0,0 +1,205 @@
+#
```
[the remark this answers is not in the document]
Comment (craiglagegit) [reply python/lsst/ip/isr/photodiodeCorrection.py:1]: I think I fixed this.

File: python/lsst/ip/isr/photodiodeCorrection.py [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+    ------
+    RuntimeError :
+        Raised if the supplied table is not 2D, or if the table has fewer
+        columns than rows (indicating that the indices are swapped).
+
+    Notes
+    -----
+    The photodiode correction attributes stored are:
+
+    abscissaCorrections : 'dict' 'float'
```
[the remark this answers is not in the document]
Comment (craiglagegit) [reply python/lsst/ip/isr/photodiodeCorrection.py:52]: I fixed this.

File: python/lsst/ip/isr/photodiodeCorrection.py [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+        Raises
+        ------
+        RuntimeError :
+            Raised if the supplied dictionary is for a different
+            calibration type.
+        """
+        calib = cls()
+
+        if calib._OBSTYPE != dictionary['metadata']['OBSTYPE']:
+            raise RuntimeError(f"Incorrect photodiode correctionsupplied.  Expected {calib._OBSTYPE}, "
```
[the remark this answers is not in the document]
Comment (craiglagegit) [reply python/lsst/ip/isr/photodiodeCorrection.py:112]: I fixed this.

File: python/lsst/ip/isr/photodiodeCorrection.py [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+
+class PhotodiodeCorrection(IsrCalib):
+    """Parameter set for linearization.
+
+    These parameters are included in cameraGeom.Amplifier, but
+    should be accessible externally to allow for testing.
+
+    Parameters
+    ----------
+    log : `logging.Logger`, optional
```
[the remark this answers is not in the document]
Comment (craiglagegit) [reply python/lsst/ip/isr/photodiodeCorrection.py:37]: I will work on this next week (week of Feb 14).

File: python/lsst/ip/isr/photodiodeCorrection.py (hunk 1 above)
Comment (craiglagegit) [reply python/lsst/ip/isr/photodiodeCorrection.py:197]: I will work on this next week (week of Feb 14).

File: python/lsst/ip/isr/photodiodeCorrection.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (craiglagegit) [reply python/lsst/ip/isr/photodiodeCorrection.py:45]: OK, i added this.

File: python/lsst/ip/isr/photodiodeCorrection.py (hunk 1 above)
Comment (craiglagegit) [reply python/lsst/ip/isr/photodiodeCorrection.py:214]: You were right, those weren't being persisted.  I fixed this in several places and created a tests/test_photodiodeCorrection.py with round trip tests, which now pass.

**Review by czwa — APPROVED** [review]

**Pull request merged**